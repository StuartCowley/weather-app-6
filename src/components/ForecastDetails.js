import PropTypes from "prop-types";
import moment from "moment";

import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;

  return (
    <section className="forecast-details">
      <article className="forecast-details__date">
        <p className="forecast-details__date__content">
          <strong>{moment(date).format("ddd Do MMM")}</strong>
        </p>
      </article>

      <article className="forecast-details__temperature">
        <p className="forecast-details__temperature__min-temp">
          Min Temp: {temperature.min} &deg;C
        </p>
        <p className="forecast-details__temperature__max-temp">
          Max Temp: {temperature.max} &deg;C
        </p>
      </article>

      <article className="forecast-details__humidity">
        <p className="forecast-details__humidity__content">
          Humidity: {humidity}%
        </p>
      </article>

      <article className="forecast-details__wind">
        <p className="forecast-details__wind__wind-speed">
          Wind Speed: {wind.speed} mph
        </p>
        <p className="forecast-details__wind__wind-direction">
          Wind Direction: {wind.direction}
        </p>
      </article>
    </section>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number.isRequired,
      max: PropTypes.number.isRequired,
    }).isRequired,
    humidity: PropTypes.number.isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number.isRequired,
      direction: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default ForecastDetails;
