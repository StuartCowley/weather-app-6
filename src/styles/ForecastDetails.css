.forecast-details {
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  max-width: 600px;
  margin: 2.5rem auto 0;
  border: 1px solid;
  text-align: center;
}
