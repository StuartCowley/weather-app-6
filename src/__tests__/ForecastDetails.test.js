import { render } from "@testing-library/react";

import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 10,
      max: 20,
    },
    humidity: 64,
    wind: {
      speed: 3,
      direction: "Test Direction",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText(/10 °C/)).toHaveClass(
      "forecast-details__temperature__min-temp"
    );
    expect(getByText(/20 °C/)).toHaveClass(
      "forecast-details__temperature__max-temp"
    );
    expect(getByText(/64%/)).toHaveClass("forecast-details__humidity__content");
    expect(getByText(/3 mph/)).toHaveClass(
      "forecast-details__wind__wind-speed"
    );
    expect(getByText(/Test Direction/)).toHaveClass(
      "forecast-details__wind__wind-direction"
    );
  });
});
