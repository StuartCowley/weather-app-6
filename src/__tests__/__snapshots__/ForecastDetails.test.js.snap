// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <section
    class="forecast-details"
  >
    <article
      class="forecast-details__date"
    >
      <p
        class="forecast-details__date__content"
      >
        <strong>
          Thu 1st Jan
        </strong>
      </p>
    </article>
    <article
      class="forecast-details__temperature"
    >
      <p
        class="forecast-details__temperature__min-temp"
      >
        Min Temp: 10 °C
      </p>
      <p
        class="forecast-details__temperature__max-temp"
      >
        Max Temp: 20 °C
      </p>
    </article>
    <article
      class="forecast-details__humidity"
    >
      <p
        class="forecast-details__humidity__content"
      >
        Humidity: 64%
      </p>
    </article>
    <article
      class="forecast-details__wind"
    >
      <p
        class="forecast-details__wind__wind-speed"
      >
        Wind Speed: 3 mph
      </p>
      <p
        class="forecast-details__wind__wind-direction"
      >
        Wind Direction: Test Direction
      </p>
    </article>
  </section>
</DocumentFragment>
`;
